clog/uwsgi_plugin.py

```diff
@@ -0,0 +1,58 @@
+import uwsgi
+import uwsgidecorators
+import logging
+import clog
+import clog.global_state
+import clog.handlers
+
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+
+
+def _mule_msg(*args, **kwargs):
+    mule = kwargs.pop('mule', 1)
+    data = pickle.dumps({
+        'service': 'uwsgi_mulefunc',
+        'func': 'log_line',
+        'args': args,
+        'kwargs': kwargs,
+    })
+    uwsgi.mule_msg(data, mule)
+
+
+class UwsgiHandler(logging.Handler):
+
+    def __init__(self, stream, mule=1):
+        logging.Handler.__init__(self)
+        self.stream = stream
+        self.mule = mule
+
+    def emit(self, record):
+        try:
+            msg = self.format(record)
+            _mule_msg(self.stream, msg, mule=self.mule)
+        except:
```

Comment:
never use bare `except:`, this catches things like `SystemExit`, `KeyboardInterrupt`, and `SystemError` which _should_ terminate the process.  If you mean to catch all exceptions, use `except Exception:`, if you explicitly mean to catch the usually-uncatchable exceptions use `except BaseException:`.  If possible, catch a specific exception type (or tuple of specific exception types)